### PR TITLE
LIB - Override registerPatient and editPatient tests with valid Liberian phone numbers

### PR DESCRIPTION
**Pull request by alaboso**
Override registerPatient and editPatient tests with valid Liberian phone numbers after the introduction of regex validation. 

@mogoodrich  - Could we update the README so that we know how to run the tests in this project on the local env?

### CODE REVIEW

File: src/test/java/org/openmrs/module/mirebalais/smoke/PatientRegistrationLiberiaFlowTest.java [hunk 1]
```diff
@@ -67,6 +120,11 @@ protected Boolean getBiometricsEnabled() {
         return true;
     }
 
+    @Override
+    protected String getContactPhoneNumber() {
+        return "+231881112233";
+    }
+
```
Comment (mogoodrich) [inline src/test/java/org/openmrs/module/mirebalais/smoke/PatientRegistrationLiberiaFlowTest.java]: The idea is that you should *only* need to override this one method to change the phone number... if you need to do more than that we're doing something wrong...

**Comment by mogoodrich** [comment]
@alaboso I tried just adding the contact phone number line and it didn't fix the test, so I'm probably missing something you are seeing... :)  I didn't debug at all though... let me know if you are able to run the tests locally based on the README instructions and if things look good.  Thanks!

**Comment by alaboso** [comment]
@mogoodrich  I thought `PatientRegistrationLiberiaFlowTest`.`registerNewPatient` calls `PatientRegistrationFlowTest`.`registerNewPatient` which calls it's (PatientRegistrationFlowTest) `getContactPhoneNumber`. There's no way we can make `PatientRegistrationFlowTest` call the sublass' `getContactPhoneNumber` if we override this in subclass.....Only if we override `registerNewPatient` in subclass and make it call overriden `getContactPhoneNumber` still within subclass then it will work.

I have not yet managed to run in my local, some dependency issues am looking into. Will update.

**Comment by mogoodrich** [comment]
@alaboso actually, even though PatientRegistrationLiberiaFlowTest calls the super class registerNewPatient, the object is stilla PatientRegistrationLiberiaFlowTest, so it still picks up any overridden methods in PatientRegistrationLiberiaFlowTest...  I did a test to confirm... I just added the getContactPhoneNumber method to PatientRegistrationLiberiaFlowTest and it successfully got through that step.

The test was still failing, but it turns out that was because it is now presenting the user with the option to manually enter an identifier... looks like you changed this in the registration app, so I assume this was intentional.  To get that to work, should just be a matter of setting the automatically enter identifier method to true... will test that now...

**Comment by mogoodrich** [comment]
So @alaboso  after adding these two changes:

https://github.com/PIH/mirebalais-smoke-tests/commit/ae6e2f50f142b78e5028a94e4a72782875cf7159

https://github.com/PIH/mirebalais-smoke-tests/commit/01623cfa60685d31c672d7be8acef47834681de8

The smoke tests make it through the registration flow... they are now failing on check-in, but I haven't looked into this in too much detail:

http://bamboo.pih-emr.org:8085/artifact/MIREBALAIS-STP/JOB1/build-5156/failed-tests-screenshots/editExistingPatient%28org.openmrs.module.mirebalais.smoke.PatientRegistrationLiberiaFlowTest%29.png

So this PR can probably close...

**Comment by brandones** [comment]
I'll take Mark's cue and close this; it's very old. Feel free to update and reopen if needed.

**Pull request closed without merging**